File: backend/app/api/v1/maneuvers.py
"""Maneuver calculation endpoints."""

import math
from fastapi import APIRouter
from app.domain.astrodynamics.models import (
    HohmannRequest, HohmannResponse, BiEllipticRequest,
    PlaneChangeRequest, ManeuverResponse,
)
from app.domain.astrodynamics.maneuvers import (
    hohmann_transfer, bielliptic_transfer, plane_change,
    combined_maneuver, tsiolkovsky_mass_ratio, circular_speed,
)

router = APIRouter()


@router.post("/hohmann", response_model=HohmannResponse)
async def hohmann(req: HohmannRequest) -> HohmannResponse:
    r = hohmann_transfer(req.r1_km, req.r2_km, req.mu)
    return HohmannResponse(**r)


@router.post("/bielliptic", response_model=ManeuverResponse)
async def bielliptic(req: BiEllipticRequest) -> ManeuverResponse:
    r = bielliptic_transfer(req.r1_km, req.rb_km, req.r2_km, req.mu)
    return ManeuverResponse(dv_total_km_s=r["dv_total_km_s"], details=r)


@router.post("/plane_change", response_model=ManeuverResponse)
async def plane_change_endpoint(req: PlaneChangeRequest) -> ManeuverResponse:
    dv = plane_change(req.v_km_s, math.radians(req.delta_i_deg))
    return ManeuverResponse(dv_total_km_s=abs(dv), details={"dv_km_s": dv})

File: frontend/src/infrastructure/api/maneuvers.ts
import { apiClient } from './client';

export interface HohmannRequest {
  r1_km: number;
  r2_km: number;
}

export interface HohmannResponse {
  dv1_km_s: number;
  dv2_km_s: number;
  dv_total_km_s: number;
  tof_s: number;
  tof_h: number;
  a_transfer_km: number;
}

export interface BiEllipticRequest {
  r1_km: number;
  rb_km: number;
  r2_km: number;
}

export interface BiEllipticResponse {
  dv1_km_s: number;
  dv2_km_s: number;
  dv3_km_s: number;
  dv_total_km_s: number;
  tof_s: number;
  tof_h: number;
  is_efficient: boolean;
}

export interface PlaneChangeRequest {
  v_km_s: number;
  delta_i_deg: number;
}

export interface PlaneChangeResponse {
  dv_km_s: number;
}

export async function calcHohmann(req: HohmannRequest): Promise<HohmannResponse> {
  const { data } = await apiClient.post<HohmannResponse>('/maneuvers/hohmann', req);
  return data;
}

export async function calcBiElliptic(req: BiEllipticRequest): Promise<BiEllipticResponse> {
  const { data } = await apiClient.post<BiEllipticResponse>('/maneuvers/bielliptic', req);
  return data;
}

export async function calcPlaneChange(req: PlaneChangeRequest): Promise<PlaneChangeResponse> {
  const { data } = await apiClient.post<PlaneChangeResponse>('/maneuvers/plane_change', req);
  return data;
}
